apps/block/components/share/index.coffee

```diff
@@ -0,0 +1,15 @@
+{ truncate } = require 'underscore.string'
+BlockShareView = require './view.coffee'
+
+generateText = ({ title }) ->
+  truncate(title, 50)
+
+module.exports = ({ block }) ->
+  $el = $('.js-block-share')
+
+  view = new BlockShareView
+    text: generateText(block.toJSON())
+
+  $el.html view.render().$el
+
+  view
```

Comment:
Woops lemme make sure this gets cleaned up in block views.

Reply:
Cleaned up as in torn down when BlockView gets removed?

Reply:
Yes. which now it does.